src/test/resources/com/puppycrawl/tools/checkstyle/javaparser/InputJavaParserNoStackOverflowOnDeepStringConcat.java

```diff
@@ -2,404 +2,6 @@
 
 public class InputJavaParserNoStackOverflowOnDeepStringConcat {
     public static final String X =
-"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+"a"+
```

Comment:
I left input unchanged on my machine and did not get a failure. Is this intended? If so, then why are we changing this input?

Reply:
The number of concatenations in this file was a delicate balance between killing the pitest mutation mentioned at https://github.com/checkstyle/checkstyle/pull/10600 but not causing StackOverflow error in `Navigator.next()`. Once we reduce the stack size, the number of concatenations needed to kill the mutation is much less; I didn't see any reason to have this input be larger than it needed to be to kill the mutation. I also thought it was a good way to show that my approach to running a test method with limited stack size would work; if it didn't, the above pitest failure would resurface.

Reply:
I saw stack overflow exception on this file in wercker recently. After  Return problem was not reproduced